lib/netext/httpext/request.go

```diff
@@ -162,9 +162,90 @@ func compressBody(algos []CompressionType, body io.ReadCloser) (io.Reader, int64
 	return buf, int64(buf.Len()), contentEncoding, body.Close()
 }
 
+func readResponseBody(
+	state *lib.State, respType ResponseType, resp *http.Response, respErr error,
+) (interface{}, error) {
+
+	if resp == nil || respErr != nil {
+		return nil, respErr
+	}
+
+	// Transperently decompress the body if it's has a content-encoding we
+	// support. If not, simply return it as it is.
+	contentEncoding := strings.TrimSpace(resp.Header.Get("Content-Encoding"))
+	if compression, err := CompressionTypeString(contentEncoding); err == nil {
+		switch compression {
+		case CompressionTypeDeflate:
+			resp.Body, respErr = zlib.NewReader(resp.Body)
+		case CompressionTypeGzip:
+			resp.Body, respErr = gzip.NewReader(resp.Body)
+		default:
+			// We have not implemented a compression ... :(
+			respErr = fmt.Errorf(
+				"unsupported compression type %s for decompression - this is a bug in k6, please report it",
+				compression,
+			)
+		}
+	}
+
+	if respType == ResponseTypeNone {
```

Comment:
Hmm actually, I should move this above the decompression - no point in wasting CPU resources decompressing stuff if we're just going to throw away the result